pkg/argocd/gitcreds.go

```diff
@@ -66,14 +67,31 @@ func getCredsFromSecret(wbc *WriteBackConfig, credentialsSecret string, kubeClie
 		}
 		return git.NewSSHCreds(string(sshPrivateKey), "", true), nil
 	} else if git.IsHTTPSURL(wbc.GitRepo) {
-		var username, password []byte
-		if username, ok = credentials["username"]; !ok {
-			return nil, fmt.Errorf("invalid secret %s: does not contain field username", credentialsSecret)
+		var username, password, githubAppID, githubAppInstallationID, githubAppPrivateKey []byte
+		if githubAppID, ok = credentials["githubAppID"]; ok {
+			if githubAppInstallationID, ok = credentials["githubAppInstallationID"]; !ok {
+				return nil, fmt.Errorf("invalid secret %s: does not contain field githubAppInstallationID", credentialsSecret)
+			}
+			if githubAppPrivateKey, ok = credentials["githubAppPrivateKey"]; !ok {
+				return nil, fmt.Errorf("invalid secret %s: does not contain field githubAppPrivateKey", credentialsSecret)
+			}
+			// converting byte array to string and ultimately int64 for NewGitHubAppCreds
+			intGithubAppID, err := strconv.ParseInt(string(githubAppID), 10, 64)
+			if err != nil {
+				return nil, err
```

Comment:
```suggestion
				return nil, fmt.Errorf("invalid value in field githubAppID: %w", err) 
```
just a tiny improvement to help with troubleshooting